.ci/utils/restore-build-cache.yml

```diff
@@ -1,101 +1,44 @@
-              ls ${ART_NAME}/_export
-              mv ${ART_NAME}/_export ${PROJECT_DIR}/_export
-            else
-              echo "No _export directory to import from build cache"
-              echo "Here's the contents of build cache:"
-              find ${BUILD_STAGINGDIRECTORY}
-            fi
-          fi
-        }
-        fetchArtifactURL
-        downloadArtifactAndContinue
+  - bash: |
+      # COMPUTE THE ESY INSTALL CACHE LOCATION AHEAD OF TIME
+      DESIRED_LEN="86"
+      # Note: This will need to change when upgrading esy version
+      # that reenables long paths on windows.
+      if [ "$AGENT_OS" == "Windows_NT" ]; then
+        DESIRED_LEN="33"
+      fi
+      HOME_ESY3="$HOME/.esy/3"
+      HOME_ESY3_LEN=${#HOME_ESY3}
+      NUM_UNDERS=$(echo "$(($DESIRED_LEN-$HOME_ESY3_LEN))")
+      UNDERS=$(printf "%-${NUM_UNDERS}s" "_")
+      UNDERS="${UNDERS// /_}"
+      THE_ESY__CACHE_INSTALL_PATH=${HOME_ESY3}${UNDERS}/i
+      if [ "$AGENT_OS" == "Windows_NT" ]; then
+        THE_ESY__CACHE_INSTALL_PATH=$( cygpath --mixed --absolute "$THE_ESY__CACHE_INSTALL_PATH")
+      fi
+      echo "THE_ESY__CACHE_INSTALL_PATH: $THE_ESY__CACHE_INSTALL_PATH"
+      # This will be exposed as an env var ESY__CACHE_INSTALL_PATH, or an
+      # Azure var esy__cache_install_path
+      echo "##vso[task.setvariable variable=esy__cache_install_path]$THE_ESY__CACHE_INSTALL_PATH"
+    displayName: '[Cache] calculate esy store path'
       
-  - powershell: esy.cmd import-dependencies
-    continueOnError: true
-    condition: and(eq(variables['AGENT.OS'], 'Windows_NT'), and(eq(variables['Build.Reason'], 'PullRequest'), and(succeeded(), ne(variables['Build.SourceBranch'], variables['System.PullRequest.TargetBranch']))))
-    displayName: "esy import-dependencies if windows (build cache from CI cache)"
-
-  - bash: esy import-dependencies
-    continueOnError: true
-    condition: and(ne(variables['AGENT.OS'], 'Windows_NT'), and(eq(variables['Build.Reason'], 'PullRequest'), and(succeeded(), ne(variables['Build.SourceBranch'], variables['System.PullRequest.TargetBranch']))))
-    displayName: "esy import-dependencies if not windows (build cache from CI cache)"
-    
-  # Remove as soon as https://github.com/esy/esy/pull/969 is resolved.
-  # For now, windows won't use build cache for problematic package.
-  - task: Bash@3
-    continueOnError: true
-    condition: and(eq(variables['AGENT.OS'], 'Windows_NT'), and(eq(variables['Build.Reason'], 'PullRequest'), and(succeeded(), ne(variables['Build.SourceBranch'], variables['System.PullRequest.TargetBranch']))))
-    displayName: 'Remove ocamlfind prebuilts if on Windows'
+  - task: Cache@2
     inputs:
-      targetType: 'inline' # Optional. Options: filePath, inline
-      script: |
-        ls ${ESY__CACHE_INSTALL_PATH}
-        rm -r ${ESY__CACHE_INSTALL_PATH}/opam__s__ocamlfind*
+      key: esy | $(Agent.OS) | esy.lock/index.json
```

Comment:
It might make sense to add a `v1` or something like that since we can't bust the cache in any other way than changing that key, or wait 7 days

Reply:
Ooh, great point. Added! ✅  :)